abnf/odata-aggregation-abnf.txt

```diff
@@ -132,13 +132,20 @@ pathSegment     = ( complexProperty / complexColProperty ) [ "/" qualifiedComple
 
 groupingProperty = grpPropPathPrefix
                    ( entityNavigationProperty [ "/" qualifiedEntityTypeName ]
-                   / primitiveProperty 
+                   / entityAnnotationInQuery  [ "/" qualifiedEntityTypeName ]
+                   / primitiveProperty        [ "/" grpPropAnnoSegment ]
                    / complexProperty

```

Comment:
Why don't we allow a type-cast here, similar to `entityNavigationProperty`? We do allow it for path segments.

Reply:
Type casts are in the `grpPropPathSegment` rule because they make sense only if a property from the casted type follows. The exception is casting a navigation property as in
```
Items?$apply=groupby((Product/self.AgricultaralProduct))
```
because in such cases an `/@odata.id` follows *implicitly*.